### PR TITLE
avoids some unnecessary casts

### PR DESCRIPTION
**Pull request by lemire**
Relates to https://github.com/simdutf/simdutf/issues/581

### CODE REVIEW

File: src/implementation.cpp [hunk 1]
```diff
@@ -3,6 +3,11 @@
 #include <climits>
 #include <type_traits>
 
+static_assert(sizeof(uint8_t) == sizeof(char), "simdutf requires that uint8_t be a char");
+static_assert(sizeof(uint16_t) == sizeof(char16_t), "simdutf requires that char16_t be 16 bits");
+static_assert(sizeof(uint32_t) == sizeof(char32_t), "simdutf requires that uint8_t be 32 bits");
```
Comment (pauldreik) [inline src/implementation.cpp:8]: copy paste typo: uint32_t

File: src/implementation.cpp [hunk 2]
```diff
@@ -3,6 +3,11 @@
 #include <climits>
 #include <type_traits>
 
+static_assert(sizeof(uint8_t) == sizeof(char), "simdutf requires that uint8_t be a char");
```
Comment (pauldreik) [inline src/implementation.cpp:6]: isn't this implicitly covered by the test on row 9 for CHAR_BIT ?

**Review by pauldreik — CHANGES_REQUESTED** [review]
would be interesting to see if this makes the freebsd unit test problem go away.

File: src/implementation.cpp (hunk 1 above)
Comment (lemire) [reply src/implementation.cpp:8]: thanks!

File: src/implementation.cpp (hunk 2 above)
Comment (lemire) [reply src/implementation.cpp]: Yes. It is for debugging purposes. I am adding a comment.

I am concerned with broken compiler systems. I want to catch them.

**Review by pauldreik — APPROVED** [review]

**Pull request merged**